ProtoGame2D/Code/Game/Framework/Game.cpp

```diff
@@ -1,25 +1,103 @@
 #include "Game.h"
 
+#include "Engine/Core/GameCommon.h"
+#include "Engine/Graphics/Renderer.h"
+#include "Engine/Graphics/Shader.h"
+#include "Engine/Graphics/ShaderDefinition.h"
+#include "Engine/Graphics/Buffers/FrameBuffer.h"
+#include "Engine/Input/InputSystem.h"
+#include "Engine/Input/WinKeys.h"
+
+float x = 512.0f;
+
+extern InputSystem* g_InputSystem;
+
 Game::Game()
 {
+	m_ShaderDef = new ShaderDefinition(*ShaderDefinition::InitializeDef(APEX_SHADER_XML));
+	m_Shader = Renderer::GetInstance()->GetOrCreateShader(m_ShaderDef);
+	
+	m_ScreenShaderDef = new ShaderDefinition(*ShaderDefinition::InitializeDef(APEX_SCREENSHADER_XML));
+	m_ScreenShader = Renderer::GetInstance()->GetOrCreateShader(m_ScreenShaderDef);
+	
+	Renderer::GetInstance()->EnableBlend(ParseBlendFac[m_Shader->GetSRC()], ParseBlendFac[m_Shader->GetDST()], ParseBlendOp[m_Shader->GetOP()]);
+
+	m_SrcBuffer = new FrameBuffer();
+	m_DestBuffer = new FrameBuffer();
 }
 
 Game::~Game()
 {
+	delete m_ShaderDef;
+	m_ShaderDef = nullptr;
+	
+	delete m_Shader;
+	m_Shader = nullptr;
+	
+	delete m_ScreenShaderDef;
+	m_ScreenShaderDef = nullptr;
+	
+	delete m_ScreenShader;
+	m_ScreenShader = nullptr;
+	
+	delete m_SrcBuffer;
+	m_SrcBuffer = nullptr;
+	
+	delete m_DestBuffer;
+	m_DestBuffer = nullptr;
 }
 
 void Game::BeginFrame()
 {
+	m_SrcBuffer->Bind();
+
+	Renderer::GetInstance()->ClearColor();
+	Renderer::GetInstance()->Clear();
+
+	m_Shader->Bind();
+
+	Mat4 camera = Mat4::orthographic(APEX_WINDOW_DIMS.m_X, APEX_WINDOW_DIMS.m_Y, APEX_WINDOW_DIMS.m_Z, APEX_WINDOW_DIMS.m_W, -2.0f, 2.0f);
+
+	m_Shader->SetUniformMat4f("u_Proj", camera);
 }
 
-void Game::Update()
+void Game::Update(float deltaseconds)
 {
+	if(g_InputSystem->IsHeldDown(D))
+	{
+		x += 10.0f * deltaseconds;
+	}
+	else if(g_InputSystem->IsHeldDown(A))
+	{
+		x -= 10.0f * deltaseconds;
+	}
+
+	if(x >= 924.0f)
```

Comment:
@anishvabardhan All this X stuff is test code to see if input system is working as you thought it would I am guessing?

Reply:
yup just for testing